### ISSUE TITLE
Hope Kvrocks can support full commands of the stream data type

### ISSUE THREAD

**Issue by jcy1001** [apache/kvrocks#1544]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/kvrocks/issues) and found no similar issues.

### Motivation

Same as the title

### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR!

**Comment by torwig** [comment]
@jcy1001 I would like to help you with your question/request/issue. But unfortunately, I don't know the language and Google Translate said `I hope to increase the stream function of redis and realize the simple mq function` which I didn't fully understand.

**Comment by jcy1001** [comment]
hi, redis has stream function in t_stream.c , redis7.0  has xgroupCreate like command , but  our project not implemented, i only find xadd xdel xinfo ..in cmd_stream.cc,  thanks ~

**Comment by jcy1001** [comment]
this is our project 

REDIS_REGISTER_COMMANDS(MakeCmdAttr<CommandXAdd>("xadd", -5, "write", 1, 1, 1),
                        MakeCmdAttr<CommandXDel>("xdel", -3, "write", 1, 1, 1),
                        MakeCmdAttr<CommandXLen>("xlen", -2, "read-only", 1, 1, 1),
                        MakeCmdAttr<CommandXInfo>("xinfo", -2, "read-only", 0, 0, 0),
                        MakeCmdAttr<CommandXRange>("xrange", -4, "read-only", 1, 1, 1),
                        MakeCmdAttr<CommandXRevRange>("xrevrange", -2, "read-only", 1, 1, 1),
                        MakeCmdAttr<CommandXRead>("xread", -4, "read-only", 0, 0, 0),
                        MakeCmdAttr<CommandXTrim>("xtrim", -4, "write", 1, 1, 1),
                        MakeCmdAttr<CommandXSetId>("xsetid", -3, "write", 1, 1, 1))

this is redis 
[XACK](http://www.redis.cn/commands/xack.html)
[XADD](http://www.redis.cn/commands/xadd.html)
[XCLAIM](http://www.redis.cn/commands/xclaim.html)
[XDEL](http://www.redis.cn/commands/xdel.html)
[XGROUP](http://www.redis.cn/commands/xgroup.html)
[XINFO](http://www.redis.cn/commands/xinfo.html)
[XLEN](http://www.redis.cn/commands/xlen.html)
[XPENDING](http://www.redis.cn/commands/xpending.html)
[XRANGE](http://www.redis.cn/commands/xrange.html)
[XREAD](http://www.redis.cn/commands/xread.html)
[XREADGROUP](http://www.redis.cn/commands/xreadgroup.html)
[XREVRANGE](http://www.redis.cn/commands/xrevrange.html)
[XTRIM](http://www.redis.cn/commands/xtrim.html)

**Comment by Yangsx-1** [comment]
We are implementing features to be compatible with Redis 7.0, it may need some time. I can drive this task when i'm free.

**Comment by torwig** [comment]
@jcy1001 Yes, you are right, currently, consumer groups aren't implemented in Kvrocks.

**Comment by jcy1001** [comment]
@Yangsx-1 @torwig  nice job ~
